File: data-r2dbc-repositories/src/main/java/com/example/demo/DataInitializer.java
package com.example.demo;

import io.r2dbc.postgresql.codec.Json;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.context.event.ContextRefreshedEvent;
import org.springframework.context.event.EventListener;
import org.springframework.r2dbc.core.DatabaseClient;
import org.springframework.stereotype.Component;

@Component
@Slf4j
@RequiredArgsConstructor
public class DataInitializer {

    private final DatabaseClient databaseClient;

    @EventListener(value = ContextRefreshedEvent.class)
    public void init() throws Exception {
        log.info("start data initialization...");
        this.databaseClient
                .sql("INSERT INTO  posts (title, content) VALUES (:title, :content)")
                .filter((statement, executeFunction) -> statement.returnGeneratedValues("id").execute())
                .bind("title", "my first post")
                .bind("content", "content of my first post")
                .fetch()
                .first()
                .subscribe(
                        data -> log.info("inserted data : {}", data),
                        error -> log.info("error: {}", error)
                );

    }
}

File: komapper/src/main/kotlin/com/example/demo/DataInitializer.kt
package com.example.demo

import com.example.demo.Post.Status.PUBLISHED
import org.komapper.core.dsl.Meta
import org.komapper.core.dsl.QueryDsl
import org.komapper.core.dsl.query.first
import org.komapper.r2dbc.R2dbcDatabase
import org.slf4j.LoggerFactory
import org.springframework.boot.context.event.ApplicationReadyEvent
import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component

@Component
class DataInitializer(
    private val r2dbcDatabase: R2dbcDatabase
) {
    companion object {
        private val log = LoggerFactory.getLogger(DataInitializer::class.java)
    }

    @EventListener(ApplicationReadyEvent::class)
    suspend fun init() {
        val p = Meta.post

        r2dbcDatabase.withTransaction {

            // create schema
            r2dbcDatabase.runQuery {
                QueryDsl.drop(p)
                QueryDsl.create(p)
            }

            // insert data
            val newPost = r2dbcDatabase.runQuery {
                QueryDsl.insert(p)
                    .single(
                        Post(
                            title = "Hello, Komapper!",
                            body = "Hello, Komapper!",
                            status = PUBLISHED
                        )
                    )
            }

            // select data
            val foundPost = r2dbcDatabase.runQuery {
                QueryDsl.from(p).where { p.id eq newPost.id }.first()
            }

            log.debug(" found post: {}", foundPost)
        }
    }
}